Formula/llvm.rb

```diff
@@ -42,14 +32,13 @@ class Llvm < Formula
   uses_from_macos "zlib"
 
   on_linux do
+    depends_on "glibc" if Formula["glibc"].any_version_installed?
     depends_on "pkg-config" => :build
     depends_on "binutils" # needed for gold
     depends_on "libelf" # openmp requires <gelf.h>
-  end
 
-  depends_on "gcc" unless OS.mac?
-
-  fails_with gcc: "5"
+    patch :DATA
```

Comment:
Maybe leave a comment so folks know where this patch came from and how to update it if needed?

```suggestion
    # Apply patches slated for the 12.0.x release stream
    # to allow building with GCC 5 and 6. Upstream bug:
    # https://bugs.llvm.org/show_bug.cgi?id=50732
    patch :DATA
```

Reply:
This comment will be very helpful when we upstream to homebrew/core and I'll add it once I get a green CI from the current run.